Cargo.toml

```diff
@@ -9,4 +9,8 @@ readme = "README.md"
 
 [dependencies]
 num-traits = "0.2.*"
-smallvec = "0.6.*"
+serde = { version = "1.0.*", optional = true }
+smallvec = { version = "0.6.*" }
+
+[features]
+serde = ["dep:serde", "smallvec/serde"]
```

Comment:
Ahh hold on this syntax requires Rust 1.60.0, will see what can be done to support older versions.

Reply:
https://github.com/rust-lang/api-guidelines/discussions/180
I'll have a play around with some of the suggestions here later.

Reply:
I have now changed it to be compatible with old versions of Rust.